R/process_tests_delayed_discharges.R

```diff
@@ -0,0 +1,53 @@
+#' Process Delayed Discharges tests
+#'
+#' @description Takes the processed Delayed Discharges extract and produces
+#' a test comparison with the previous data. This is written to disk as a CSV.
+#'
+#' @param data a [tibble][tibble::tibble-package] of the processed data extract.
+#' @param year the financial year of the extract in the format '1718'.
+#'
+#' @return a [tibble][tibble::tibble-package] containing a test comparison.
+#'
+#' @export
+process_tests_delayed_discharges <- function(data, year) {
+  old_data <- get_existing_data_for_tests(data)
+
+  comparison <- produce_test_comparison(
+    old_data = produce_source_dd_tests(old_data),
+    new_data = produce_source_dd_tests(data)
+  ) %>%
+    write_tests_xlsx(sheet_name = "DD", year)
+
+  return(comparison)
+}
+
+#' Delayed Discharges extract tests
+#'
+#' @description Produce tests for the delayed discharges extract.
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path()])
+#'
+#' @return a dataframe with a count of each flag
+#' from [calculate_measures()]
+#'
+#' @family extract test functions
+#' for creating test flags
+#' @seealso calculate_measures
+produce_source_dd_tests <- function(data) {
+  test_flags <- data %>%
+    dplyr::mutate(
+      n_delay_episodes = 1L,
+      code9_episodes = dplyr::case_when(
+        primary_delay_reason == "9" ~ 1L,
+        TRUE ~ 0L
+      )
```

Comment:
```suggestion
      code9_episodes = dplyr::if_else(.data$primary_delay_reason == "9", 1L, 0L)
```

Reply:
I think that actually, this will be sufficient.
```suggestion
      code9_episodes = .data$primary_delay_reason == "9"
```

As the `TRUE`/`FALSE` will get summed as though it's `1`/`0`. Similar to #689